api/t8_fortran_interface/t8_fortran_interface.c

```diff
@@ -0,0 +1,193 @@
+/*
+  This file is part of t8code.
+  t8code is a C library to manage a collection (a forest) of multiple
+  connected adaptive space-trees of general element classes in parallel.
+
+  Copyright (C) 2024 the developers
+
+  t8code is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2 of the License, or
+  (at your option) any later version.
+
+  t8code is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with t8code; if not, write to the Free Software Foundation, Inc.,
+  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+*/
+
+#include <api/t8_fortran_interface/t8_fortran_interface.h>
+#include <t8_forest/t8_forest_general.h>
+#include <t8_forest/t8_forest_geometrical.h>
+#include <t8_cmesh/t8_cmesh_examples.h>
+#include <t8_cmesh/t8_cmesh_helpers.h>
+#include <t8_element_c_interface.h>
+#include <t8_schemes/t8_default/t8_default_c_interface.h>
+
+void
+t8_fortran_init_all_ (sc_MPI_Comm * comm)
+{
+  T8_ASSERT (comm != NULL);
+  /* Initialize sc */
+  sc_init (*comm, 1, 1, NULL, SC_LP_DEFAULT);
+  /* Initialize t8code */
+  t8_init (SC_LP_DEFAULT);
+}
+
+/* Wrapper around sc_finalize */
+void
+t8_fortran_finalize ()
+{
+  sc_finalize ();
+}
+
+void t8_fortran_cmesh_commit(t8_cmesh_t cmesh, sc_MPI_Comm* comm){
+	t8_cmesh_commit(cmesh, *comm);
+}
+
+void t8_fortran_cmesh_set_join_by_stash_noConn(t8_cmesh_t cmesh, const int do_both_directions){
+	t8_cmesh_set_join_by_stash(cmesh, NULL, do_both_directions);
+}
+
+void
+t8_fortran_init_all (sc_MPI_Comm * comm)
+{
+  int                 rank;
+
+  t8_fortran_init_all_ (comm);
+  if (*comm != sc_MPI_COMM_NULL) {
+    sc_MPI_Comm_rank (*comm, &rank);
+    t8_debugf ("rank = %i\n", rank);
+  }
+
+}
+
+void
+t8_fortran_init_all_noMPI ()
+{
+  sc_MPI_Comm         commnull = sc_MPI_COMM_NULL;
+  t8_fortran_init_all (&commnull);
+}
+
+/* Build C MPI comm from Fortran MPI Comm. */
+sc_MPI_Comm        *
+t8_fortran_MPI_Comm_new (MPI_T8_Fint Fcomm)
+{
+#if !T8_ENABLE_MPI
+  SC_ABORT ("t8code was not configured with MPI support.");
+  return NULL;
+#endif
+  /* We use malloc instead of T8_ALLOC since t8code may not be initialized
+   * yet. */
+  sc_MPI_Comm        *Ccomm = (sc_MPI_Comm *) malloc (sizeof (*Ccomm));
+  *Ccomm = MPI_Comm_f2c (Fcomm);

```

Comment:
```suggestion
  #ifdef T8_ENABLE_MPI
  /* If configured with MPI, transform the Fortran communicator handle to a C handle */
  *Ccomm = MPI_Comm_f2c (Fcomm);
  #else
  /* In case it is not configured with MPI, set the communicator to NULL as a fallback */
  *Ccomm = sc_MPI_COMM_NULL;
  #endif
```
The only sustaining problem is that the function `MPI_Comm_f2c` call issues a warning of "implicit declaration" when the code is configured/compiled without MPI and without "--enable-fortran", since the headers does not include `mpi.h` when SC and t8code is configured without MPI. In order to catch this case, we need to have a fallback similar to the suggestion above.